Bottom: 10ca7f91fd534938ec457af819225a3d34941088
Top:    da1931616aae55bff70bdf9b2939e314d21513d7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-06 17:35:08 +0000

Hack around Circuit CAM / EDG brokeness reading drill files








---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 54c7f4a..dfa7f00 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -52,10 +52,16 @@ static void gerber_fill_polygon (hidGC gc, int n_coords, int *x, int *y);
 #define gerberYOffset(pcb, y) ((long) (-(y)))
 
 /* These are for drills */
-#define gerberDrX(pcb, x) ((long) ((x)/10))
-#define gerberDrY(pcb, y) ((long) (((pcb)->MaxHeight - (y)))/10)
-#define gerberDrXOffset(pcb, x) ((long) ((x)/10))
-#define gerberDrYOffset(pcb, y) ((long) (-(y))/10)
+#ifdef EDG_ARE_BROKEN
+#  define EDG_MUL 100
+#else
+#  define EDG_MUL 1
+#endif
+
+#define gerberDrX(pcb, x) ((long) (EDG_MUL * (x)/10))
+#define gerberDrY(pcb, y) ((long) (EDG_MUL * ((pcb)->MaxHeight - (y))/10))
+#define gerberDrXOffset(pcb, x) ((long) (EDG_MUL * (x) / 10))
+#define gerberDrYOffset(pcb, y) ((long) (EDG_MUL * -(y) / 10))
 
 /*----------------------------------------------------------------------------*/
 /* Private data structures                                                    */
